packages/client-common/src/client.ts

```diff
@@ -229,10 +249,21 @@ export class ClickHouseClient<Stream = unknown> {
    */
   async insert<T>(params: InsertParams<Stream, T>): Promise<InsertResult> {
     const format = params.format || 'JSONCompactEachRow'
-
     this.valuesEncoder.validateInsertValues(params.values, format)
-    const query = `INSERT INTO ${params.table.trim()} FORMAT ${format}`
 
+    let columnsPart = ''
+    if (params.columns !== undefined) {
```

Comment:
optional nit: We could consider extractimg in a separate function to simplify testing:
```js
function getColumnsStatement(columns) {
  if (!columns) return '';
  if (Array.isArray(columns) && params.columns.length > 0) {
        return (${params.columns.join(',')})`
  if (
        isInsertColumnsExcept(params.columns) &&
        params.columns.except.length > 0
      ) {
        return` (* EXCEPT (${params.columns.except.join(',')}))`
      }
    }
}
```